src/typing/type_inference_js.ml

```diff
@@ -970,7 +965,7 @@ and statement_decl cx = Ast.Statement.(
       let _, { Ast.Identifier.name; _ } = id in
       let r = mk_reason (spf "class %s" name) loc in
       let tvar = Flow_js.mk_tvar cx r in
-      Env_js.init_env cx name (create_env_entry tvar tvar (Some loc))
+      Env_js.init_env cx name (create_env_entry tvar tvar (Some loc)) VarScope (* TODO: hoist behavior *)
```

Comment:
@jeffmo I'm fairly sure that declared classes should have lexical scoping rules, but what about interfaces? There's no guidance from ES6, but I think the spirit of the decision to make classes lexically scope also argues for the same behavior in interfaces.